src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -214,7 +217,16 @@ internal static class CSharpDiagnosticFilter
 
             if (!isSpecified)
             {
-                report = isEnabledByDefault ? ReportDiagnostic.Default : ReportDiagnostic.Suppress;
+                if (errorCode == (int)ErrorCode.WRN_Experimental)
+                {
+                    // Special handling for [Experimental] warning (treat as error severity by default)
```

Comment:
```suggestion
                    // Special handling for [Experimental] warning (treat as error severity by default)
                    Debug.Assert(isEnabledByDefault);
                    Debug.Assert(!specifiedWarnAsErrorMinus);
```

Both of those should hold correct?